pkg/unions/kubernetes_ingress.go

```diff
@@ -43,3 +44,64 @@ func NewKubernetesIngressFromParams(ingressParams generator.IngressParams) *Kube
 func (ki *KubernetesIngress) IsGenerated() bool {
 	return ki.isGenerated
 }
+
+func (ki *KubernetesIngress) GetName() string {
```

Comment:
Please add comments for this and the functions below

Reply:
Adding